### DISCUSSION TITLE
re-frame doesn't work on mobile?

### DISCUSSION THREAD

**Discussion by wqhhust** [day8/re-frame#702] [Q&A]
I have post the following link

http://stackoverflow.com/questions/32285936/regent-in-clojurescript-selected-value-could-be-save-in-pc-browser-but-cant-b
Using re-frame, the app works on PC but won't work on mobile, so I am wondering whether re-frame same as re-com and don't support mobile.
To verify it, just clone the following, 
https://github.com/wqhhust/test_frame/
lein clean;lein figwheel dev
and in browser, lick the first element, on the modal window pick up a feedback and save, then re-click the same element to open the modal window:  the difference is when re-click, in PC browser I could see the saved feedback, but in mobile browser the saved feedback is gone.

**Comment by mike-thompson-day8** [comment]
I'm seeing odd looking code:
https://github.com/wqhhust/test_frame/blob/f2e04cf3fa75f599eb35e816c02b6d7ee5156fbf/src/cljs/test_frame/views.cljs#L18-L38

`new-feedback` is a ratom created from a prop,  but the moment that ratom gets `reset!`, the entire view will rerender, which will simply recreate the ratom with the value of the prop.  Doesn't make any sense, at all. Did you mean to make this a Form2 Component?  

Ether way, this not the place to be asking for help writing your code.   This is the place for reporting issues with re-frame.

**Comment by plexus** [comment]
> Ether way, this not the place to be asking for help writing your code. This is the place for reporting issues with re-frame.

If you're interested to have a mailing list or forum for these kind of questions then have a look at http://clojureverse.org. It's a free, non commercial alternative set up precisely with that purpose.

**Comment by mike-thompson-day8** [comment]
There are an overwhelming  number of channels out there.  I tend to look at:
-   slack clojurians http://clojurians.net/    (prefered)  (re-frame or reagent channels)
-   google groups  https://groups.google.com/forum/#!forum/clojurescript

**Comment by wqhhust** [comment]
Thanks mike-thompson-day8  for reviewing the code,
and unfortunately I can't access both google and clojurians.net since I am in China and these sites are blocked.

Actually my original code works for PC browser. The issue is that it doesn't work on mobile browser. I attached 5 snapshot, the last is works for PC browser, but won't work for  mobile.

The logic is simple:  click an item, it shows a modal and then pick up a feedback from a list and then save, when show the same item again, the pickup data should still exists, but it is gone for mobile browser, while it's there on PC browser.

Looks like I don't need Form 2 component: I don't need to re-render when pick up a new option which will reset the atom, the reset is just store the picked up value, so that when click the save button, the data will be saved into the global db. I also tried to use Form 2 component, it still doesn't work on mobile.

![1](https://cloud.githubusercontent.com/assets/5955201/9566780/e9216ccc-4f44-11e5-8cf0-10aed1d0e50d.png)

![2](https://cloud.githubusercontent.com/assets/5955201/9566782/f2e6298c-4f44-11e5-960b-a33b0b652b77.png)

![3](https://cloud.githubusercontent.com/assets/5955201/9566785/f7417b3a-4f44-11e5-9579-a067beabc9ee.png)

![4](https://cloud.githubusercontent.com/assets/5955201/9566787/fbd659ae-4f44-11e5-9168-9c3a4d9ef534.png)

![5](https://cloud.githubusercontent.com/assets/5955201/9566789/00c9e052-4f45-11e5-9f5c-2d09873b614e.png)

**Comment by mike-thompson-day8** [comment]
I'm not aware of any reason for mobile and desktop to be different.  As I said above, I can see problems in your existing code.
